lib/galaxy/webapps/galaxy/services/histories.py

```diff
@@ -640,11 +641,7 @@ def get_custom_builds_metadata(
         installed_builds = []
         for build in glob.glob(os.path.join(trans.app.config.len_file_path, "*.len")):
             installed_builds.append(os.path.basename(build).split(".len")[0])
-        fasta_hdas = (
-            trans.sa_session.query(model.HistoryDatasetAssociation)
-            .filter_by(history=history, extension="fasta", deleted=False)
-            .order_by(model.HistoryDatasetAssociation.hid.desc())
-        )
+        fasta_hdas = get_fasta_hdas_by_history(trans.sa_session, history.id)
```

Comment:
seems easier to read inline and isn't used anywhere else, so maybe just keep it here ? otherwise bind it to the History class ?

Reply:
I think it would be better to move low-level SA code out of services and into a dedicated location (class, module, package - whatever). In my opinion, it's easy to read it inline when one has read it many times, or have read such code many times; otherwise one has to switch conceptual context and go from business logic to "what exactly does this retrieve from the database". Having a clearly named method/function instead eliminates this step (get HDAs by history), thus reducing cognitive load. One reads code faster that way, I think.

As for `model.History`: I think it would be great if we could limit the model classes to model definitions only + and definitions and functionality related to model instances. This function is fetching related data from the database - I think we should place it elsewhere. Since we haven't agreed yet on what to do with the repositories idea, I think putting them into appropriate manager modules as lightweight standalone functions (also, as per @jmchilton's suggestion) is a reasonable compromise?

Reply:
>  I think though just a static method on the HDA class or a function in that module that took in the sa_session along with the rest of the arguments you've created and just did the work directly would be simpler and feel more "python" to me.

?

Reply:
this is never going to be used anywhere else, so having it at least in this module is an upside, since it's got nothing to do with the manager

Reply:
Done.

Yes, John meant the model class, I was wrong about that. However, I do think this is low-level code that belongs in a separate location, not all over the code base - for the reasons I've outlined above. In the absence of a dedicated repository, a manager seems the next best thing to me.